### ISSUE TITLE
Abort multiple devices are used with quick-mode

### ISSUE THREAD

**Issue by sdir** [ntop/PF_RING#745]
https://github.com/ntop/PF_RING/blob/e16437551f4cba5cef12135e7119b8ab844504ee/kernel/pf_ring.c#L5398

ring_release only change `device_rings` with `ring_dev`, other bind dev not clean.

**Comment by cardigliano** [comment]
@sdir could you comment more on this and better explain what you mean? Thank you.

**Comment by sdir** [comment]
like 
```sh
# first 
pfdump -i eth0,eth1 

# second
pfdump -i eth1 
```
The first command can be executed successfully, the second command fails because it cannot be bound.

**Comment by cardigliano** [comment]
@sdir this is a limitation of quick mode, it allows a single socket per interface/queue

**Comment by sdir** [comment]
Means it can't be used like this? Thank you for your answer

**Comment by cardigliano** [comment]
Correct, you cannot run 2 applications on the same interface if quick mode is enabled

**Comment by sdir** [comment]
Not running at the same time, after the first command ends, the second one cannot be executed. eth1 can't pfring_open forever. the interface eth1 device_rings not set to null

**Comment by cardigliano** [comment]
I pushed a fix for this, please update from the git dev branch and let me know

**Comment by sdir** [comment]
Would it be better to write like this
```c
  if (pfr->ring_dev->dev && pfr->ring_dev == &any_device_element)
    num_any_rings--;
  else
  {
    u_int64_t channel_id_bit = 0;
    int i = 0;
    /* Check all bound devices in case of multi devices */
    list_for_each_safe(ptr, tmp_ptr, &ring_aware_device_list)
    {
      pf_ring_device *dev_ptr = list_entry(ptr, pf_ring_device, device_list);
      int32_t dev_index = ifindex_to_pf_index(dev_ptr->dev->ifindex);
      if (dev_index >= 0 && test_bit(dev_index, pfr->pf_dev_mask))
      {
        if (num_rings_per_device[dev_index] > 0)
          num_rings_per_device[dev_index]--;

        if (quick_mode) {
          /* Reset quick mode for all channels */
          for (i = 0; i < MAX_NUM_RX_CHANNELS; i++)
          {
            channel_id_bit = 1 << i;
            if ((pfr->channel_id_mask & channel_id_bit) && quick_mode_rings[dev_index][i] == pfr)
              quick_mode_rings[dev_index][i] = NULL;
          }
        }
      }
    }
  }
```

**Comment by cardigliano** [comment]
Are you saying that there is a bug also in the num_rings_per_device counters in case of multiple devices? Let me check

**Comment by cardigliano** [comment]
Fixed in e57a3adbe846233ba4df06a52c79e4bb17f79015
